packages/strapi-plugin-users-permissions/config/policies/permissions.js

```diff
@@ -42,18 +38,33 @@ module.exports = async (ctx, next) => {
   }
 
   // Retrieve `public` role.
-  if (!role) {
-    role = await strapi.query('role', 'users-permissions').findOne({ type: 'public' }, []);
+  if (roles.length == 0) {
+    roles.push(await strapi.query('role', 'users-permissions').findOne({ type: 'public' }, []));
   }
 
   const route = ctx.request.route;
-  const permission = await strapi.query('permission', 'users-permissions').findOne({
-    role: role._id || role.id,
-    type: route.plugin || 'application',
-    controller: route.controller,
-    action: route.action,
-    enabled: true
-  }, []);
+  let bandera = false;
+  let permission;
+  for (const role of roles) {
+    if (role.type === 'root') {
+      return await next();
+    }
+    permission = await strapi.query('permission', 'users-permissions').findOne({
+      role: role._id || role.id,
+      type: route.plugin || 'application',
+      controller: route.controller,
+      action: route.action,
+      enabled: true
+    }, []);
+    if (permission) {
+      bandera = true;
+      break;
+    }
+  }
+  if (bandera == false) {
```

Comment:
Assuming you mean flag or something of the sort here. You may want to make sure you are using english variable names here.